lib/locales.js

```diff
@@ -0,0 +1,18 @@
+import { languages, namespaces } from "next-i18next-static-site";
+
+// Load all locales, required for next-i18n-static-site
+// Can be changed based on your folder structure
+const locales = {};
+languages.map((language) => {
+  locales[language] = {};
+
+  namespaces.map((namespace) => {
+    locales[language][namespace] = require("./../locales/" +
+      language +
+      "/" +
+      namespace +
+      ".json");
+  });
+});
```

Comment:
Kindly wrap this in a function and then return the required response from the function.

Reply:
Actually, this is not my implementation. It is suggested by the plugin author since were in need to use the static site built due to Netlify restrictions.

Reply:
Can you please share the source you are following to create this file?

IMHO, even if we have to implement such logic, I don't think it will make any difference while implementing it inside functions. Reason why I'm asking for making it as a function is that we should work on code quality as well, instead of just writing it as plain implementation of js. Also, it will be easier to add unit tests (later on) for this file.

Reply:
Done! I totally agree with your points.

I am following [this package](https://github.com/xairoo/next-i18next-static-site). Their [one of the examples](https://github.com/xairoo/next-i18next-static-site/tree/main/examples/web) is what I tried imitating in our project.

The reason to use this package is that previously I was using the next-i18next package to handle the translation-related logic, and everything was going smoothly inside the local dev. environment.
As I pushed that code, it became useless because Netlify cannot process translations on the server, and need pre-generated translations.
Another option was to use a Node.JS environment to host the website, but that doesn't seem viable.